restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockMediaGateway.java

```diff
@@ -110,55 +117,62 @@
     private ActorRef monitoringService;
 
     private File recordingFile = null;
+    private int recordingMaxLength;
+
+    private DateTime startTime;
+    private DateTime endTime;
 
     private ActorSystem system;
+    private NotificationRequest recordingRqnt;
```

Comment:
maybe we can refactor this on next iteration to have a map of ongoing recordingRqnt and associated sender with reqId as key, so we may notify later on any scenario (including multiple RQNY by multiple RQNT senders) ...

Currently this seems to specific to record scenario...

Reply:
@jaimecasero yes that could be needed for parallel testing indeed. I can do it now so we can enable parallel testing for this test class

Reply:
@jaimecasero actually parallel testing for this class wont be able because we rely on `ReceiveTimeout` to mock the maxLength attribute of Recording. 
In case of parallel testing, the `ReceiveTimeout` timer will never hit.

I live this for next iteration